frontend/src/components/TicketReservationForm/index.jsx

```diff
@@ -0,0 +1,207 @@
+import { gender, specialNeeds, mealSpecification } from "./validation";
+import { countries, countryCodes } from "../signup/SignUpForm/validation";
+import Button from "../shared/Button";
+import Input from "../shared/Input";
+import "./style.css";
+const TicketForm = ({ numberOfTickets }) => {
+  const [currentTicket, setCurrentTicket] = useState(1);
+  const [tickets, setTickets] = useState([]);
+
+  const {
+    values,
+    errors,
+    touched,
+    isSubmitting,
+    handleChange,
+    handleBlur,
+    resetForm,
+    handleSubmit,
+  } = useTicketReservationForm((submittedValues) => {
+    setTickets((prevTickets) => [...prevTickets, submittedValues]);
+    if (currentTicket < numberOfTickets) {
+      setCurrentTicket(currentTicket + 1);
+      resetForm();
+    } else {
+      ticketReservationAPI(tickets.concat(submittedValues));
+    }
+  });
+
+  return (
+    <>
+      <h1 className="ticket-reservation-heading">Ticket Reservation</h1>
+      <h2>
+        Filling details for ticket {currentTicket} of {numberOfTickets}
+      </h2>
+      <form className={`ticket-form-container`} onSubmit={handleSubmit}>
+        <div className="inputs-container">
+          <fieldset className="passport-details-container">
+            <legend>Passport Details</legend>
+            <div className="row">
+              <Input
+                label="Passport Number"
+                type="text"
+                id="passportNumber"
+                placeholder="Enter your passport number"
+                onChange={handleChange}
+                value={values.passportNumber}
+                onBlur={handleBlur}
+                showError={errors.passportNumber && touched.passportNumber}
+                errorMessage={errors.passportNumber}
+              />
+              <Input
+                label="Issuing Country"
+                type="text"
+                id="passportIssuingCountry"
+                onChange={handleChange}
+                value={values.passportIssuingCountry}
+                onBlur={handleBlur}
+                showError={
+                  errors.passportIssuingCountry &&
+                  touched.passportIssuingCountry
+                }
+                errorMessage={errors.passportIssuingCountry}
+                selectionInput
+                defaultSelectionText="Select Country"
+                options={countries}
+              />
+            </div>
+          </fieldset>
+          <fieldset className="personal-details-container">
+            <legend>Personal Details</legend>
+            <div className="row">
+              <Input
+                label="Date of Birth"
+                type="date"
+                id="dateOfBirth"
+                onChange={handleChange}
+                value={values.dateOfBirth}
+                onBlur={handleBlur}
+                showError={errors.dateOfBirth && touched.dateOfBirth}
+                errorMessage={errors.dateOfBirth}
+              />
```

Comment:
Consider setting min and max date validations for dateOfBirth to ensure realistic entries.